cmd/kubernetesDeploy.go

```diff
@@ -39,15 +39,22 @@ func kubernetesDeploy(config kubernetesDeployOptions, telemetryData *telemetry.C
 	runKubernetesDeploy(config, &c, log.Writer())
 }
 
-func runKubernetesDeploy(config kubernetesDeployOptions, command command.ExecRunner, stdout io.Writer) {
+func runKubernetesDeploy(config kubernetesDeployOptions, command command.ExecRunner, stdout io.Writer) error {
 	if config.DeployTool == "helm" || config.DeployTool == "helm3" {
-		runHelmDeploy(config, command, stdout)
-	} else {
-		runKubectlDeploy(config, command)
+		return runHelmDeploy(config, command, stdout)
+	} else if config.DeployTool == "kubectl" {
+		return runKubectlDeploy(config, command)
 	}
+	return nil
```

Comment:
Here the code should return an error and the caller should fail.

Reply:
validate the return value and fail id err!=nil
https://github.com/SAP/jenkins-library/blob/b9c8130b914a5834076ca92e0650fd3d538b8b56/cmd/kubernetesDeploy.go#L39